standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStore.java

```diff
@@ -38,10 +38,11 @@
 import org.apache.hadoop.hive.metastore.Warehouse;
 import org.apache.hadoop.hive.metastore.annotation.MetastoreCheckinTest;
 import org.apache.hadoop.hive.metastore.api.*;
-import org.apache.hadoop.hive.metastore.api.utils.DecimalUtils;
 import org.apache.hadoop.hive.metastore.client.builder.DatabaseBuilder;
 import org.apache.hadoop.hive.metastore.columnstats.cache.LongColumnStatsDataInspector;
 import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.hadoop.hive.metastore.txn.TxnStore;
```

Comment:
Add tests to check if invalidation happens only for stale data or not. Also, after refresh it should be valid.

Reply:
I will do that in new PR. Where I will put tableId and writeId check on all the methods and there corresponding tests in cacheStore. This PR is mostly around adding validWriteIdList to TableWrapper Object in cacheStore.